AlphaWallet/UI/ButtonsBar.swift

```diff
@@ -3,123 +3,201 @@
+        
+        innerStackView.addArrangedSubview(buttonsStackView)
+        
+        addSubview(innerStackView)
+        
         let margin = CGFloat(20)
         NSLayoutConstraint.activate([
-            buttonsStackView.anchorsConstraint(to: self, edgeInsets: .init(top: 0, left: margin, bottom: 0, right: margin)),
+            innerStackView.anchorsConstraint(to: self, edgeInsets: .init(top: 0, left: margin, bottom: 0, right: margin)),
         ])
     }
 
     required init?(coder aDecoder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
-
+    
+    fileprivate func setup(viewModel: ButtonsBarViewModel, view: ContainerViewWithShadow<UIButton>) {
+        view.configureShadow(color: viewModel.buttonShadowColor, offset: viewModel.buttonShadowOffset, opacity: viewModel.buttonShadowOpacity, radius: viewModel.buttonShadowRadius, cornerRadius: viewModel.buttonCornerRadius)
+
+        let button = view.childView
+        button.setBackgroundColor(viewModel.buttonBackgroundColor, forState: .normal)
+        button.setBackgroundColor(viewModel.disabledButtonBackgroundColor, forState: .disabled)
+        
+        button.setTitleColor(viewModel.buttonTitleColor, for: .normal)
+        button.setTitleColor(viewModel.disabledButtonTitleColor, for: .disabled)
+        
+        button.titleLabel?.font = viewModel.buttonFont
+        button.titleLabel?.adjustsFontSizeToFitWidth = true
+        //So long titles (that cause font to be adjusted) have some margins on the left and right
+        button.contentEdgeInsets = .init(top: 0, left: 3, bottom: 0, right: 3)
+        
+        button.cornerRadius = viewModel.buttonCornerRadius
+        button.borderColor = viewModel.buttonBorderColor
+        button.borderWidth = viewModel.buttonBorderWidth
+    }
+    
     func configure() {
-        let viewModel = ButtonsBarViewModel()
-
-        for each in buttonContainerViews {
-            each.configureShadow(color: viewModel.buttonShadowColor, offset: viewModel.buttonShadowOffset, opacity: viewModel.buttonShadowOpacity, radius: viewModel.buttonShadowRadius, cornerRadius: viewModel.buttonCornerRadius)
-
-            let button = each.childView
-            button.setBackgroundColor(viewModel.buttonBackgroundColor, forState: .normal)
-            button.setBackgroundColor(viewModel.disabledButtonBackgroundColor, forState: .disabled)
-            button.setTitleColor(viewModel.buttonTitleColor, for: .normal)
-            button.setTitleColor(viewModel.disabledButtonTitleColor, for: .disabled)
-            button.titleLabel?.font = viewModel.buttonFont
-            button.titleLabel?.adjustsFontSizeToFitWidth = true
-            //So long titles (that cause font to be adjusted) have some margins on the left and right
-            button.contentEdgeInsets = .init(top: 0, left: 3, bottom: 0, right: 3)
+        configuration.buttonViewModels.enumerated().forEach { (index, buttonType) in
```

Comment:
- [ ] Can you use this instead? More idiomatic and has a few advantages — can break/continue and not need a `self` reference.

```
for (index, buttonType) in configuration.buttonViewModels.enumerated() {
```